spec/controllers/posts_controller_spec.rb

```diff
@@ -0,0 +1,163 @@
+  end
+
+  describe "POST #create" do
+    context "with valid params" do
+      it "creates a new Post" do
+        expect {
+          post :create, {:post => valid_attributes}, valid_session
+        }.to change(Post, :count).by(1)
+      end
+
+      it "assigns a newly created post as @post" do
+        post :create, {:post => valid_attributes}, valid_session
+        expect(assigns(:post)).to be_a(Post)
+        expect(assigns(:post)).to be_persisted
+      end
+
+      it "redirects to the created post" do
+        post :create, {:post => valid_attributes}, valid_session
+        expect(response).to redirect_to('/posts')
+      end
+    end
+
+    context "with invalid params" do
+      it "assigns a newly created but unsaved post as @post" do
+        post :create, {:post => invalid_attributes}, valid_session
+        expect(assigns(:post)).to be_a_new(Post)
+      end
+
+      it "re-renders the 'new' template" do
+        post :create, {:post => invalid_attributes}, valid_session
+        expect(response).to render_template("new")
+      end
+    end
+  end
+
+  describe "PUT #update" do
+    context "with valid params" do
+      let(:new_attributes) {
+        {user_id: @user.id, content: "Updated content"}
+      }
+
+      it "updates the requested post" do
+        post = Post.create! valid_attributes
+        put :update, {:id => post.to_param, :post => new_attributes}, valid_session
+        post.reload
+        expect(post.content).to eq(new_attributes[:content])
+      end
+
+      it "assigns the requested post as @post" do
+        post = Post.create! valid_attributes
+        put :update, {:id => post.to_param, :post => valid_attributes}, valid_session
+        expect(assigns(:post)).to eq(post)
+      end
+
+      it "redirects to the post" do
+        post = Post.create! valid_attributes
+        put :update, {:id => post.to_param, :post => valid_attributes}, valid_session
+        expect(response).to redirect_to('/posts')
+      end
+    end
+
+    context "with invalid params" do
+      it "assigns the post as @post" do
+        post = Post.create! valid_attributes
+        put :update, {:id => post.to_param, :post => invalid_attributes}, valid_session
+        expect(assigns(:post)).to eq(post)
+      end
+
+      it "re-renders the 'edit' template" do
+        post = Post.create! valid_attributes
+        put :update, {:id => post.to_param, :post => invalid_attributes}, valid_session
+        expect(response).to render_template("edit")
+      end
+    end
+  end
+
+  describe "DELETE #destroy" do
+    it "destroys the requested post" do
+      post = Post.create! valid_attributes
+      expect {
+        delete :destroy, {:id => post.to_param}, valid_session
+      }.to change(Post, :count).by(-1)
+    end
+
+    it "redirects to the posts list" do
+      post = Post.create! valid_attributes
+      delete :destroy, {:id => post.to_param}, valid_session
+      expect(response).to redirect_to(posts_url)
+    end
+  end
+
```

Comment:
Again, generated tests. I cleaned it up to reduce that repetition. I had to call the variable user_post since post was getting confused with the http verb post.